src/core/PlaneWaveBasis.jl

```diff
@@ -0,0 +1,176 @@
+using FFTW
+using StaticArrays
+
+struct PlaneWaveBasis{T <: Real}
+    # Lattice and reciprocal lattice vectors in columns
+    lattice::SMatrix{3, 3, T, 9}
```

Comment:
Maybe use a `SMat3{T}` and `SVec3` typedef for this? It comes up a lot

Reply:
Good point.